### PR TITLE
Part one

### PR DESCRIPTION
**Pull request by arb**
- Added ui route
- Moved code around so the get data method is available
- Add webpack build setup
- Added stubs for UI layer
- Updated routes sort to go by plugin and then by path to keep things more together
- Related to #9

### CODE REVIEW

File: lib/setup.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+      labels: connection.settings.labels,
+      routes: connection.table().map(function (route) {
+        return {
+          method: route.method.toUpperCase(),
+          path: route.path,
+          plugin: route.public.realm.plugin || null
+        };
+      }).sort(function (a, b) {
+        // keep nulls grouped together
+        var aPlug = a.plugin || 'z';
```
Comment (cjihrig) [inline lib/setup.js:41]: This seems hacky (what if your plugin is zzz, the null would stay together, but not last in order). Personally, I think the routing table should be sorted by path. If you wanted to get fancy, you could then sort by method. I think the ability to sort by different things is a GUI feature.

File: lib/setup.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
+'use strict';
+var Hoek = require('hoek');
+
+var constructResponse = function (server) {
```
Comment (cjihrig) [inline lib/setup.js:4]: Why change this from `function constructResponse (server)`

File: test-server.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
+var Assert = require('assert');
+
```
Comment (cjihrig) [inline test-server.js:2]: No need for the extra lines

File: package.json [hunk 4]
```diff
@@ -17,14 +17,28 @@
     "hoek": "2.x.x"
   },
   "devDependencies": {
+    "autoprefixer-stylus": "0.8.0",
```
Comment (cjihrig) [inline package.json:20]: Can we use consistent notation here if possible.

File: test-server.js [hunk 5]
```diff
@@ -0,0 +1,21 @@
+var Assert = require('assert');
```
Comment (cjihrig) [inline test-server.js:1]: Can you name this file `dev-server.js`

**Comment by cjihrig** [comment]
Can you add a note in the README that using this requires installing inert too.

**Comment by arb** [comment]
After sleeping on it, this still needs some work. I'll reopen later today probably.

File: lib/setup.js (hunk 1 above)
Comment (arb) [inline lib/setup.js:41]: The nulls didn't stay together, that's why I made the change.

File: lib/setup.js (hunk 2 above)
Comment (arb) [inline lib/setup.js:4]: Consistency? I thought we used function expressions.

File: package.json (hunk 4 above)
Comment (arb) [inline package.json:20]: The consistency is:
- hapi/cl modules are `#.x.x`
- outside modules version > 1 => `#.#.x`
- outside modules version < 1 => `#.#.#`

Comment (cjihrig) [inline package.json:20]: Makes sense.

File: lib/setup.js (hunk 1 above)
Comment (cjihrig) [inline lib/setup.js:41]: I meant what would happen after these changes. Keeping the nulls together doesn't seem important. My guess is that people want to sort the routing table, not the plugins that register each route. And like I said, multiple sort criteria seems like a front end "feature".

File: lib/setup.js (hunk 2 above)
Comment (cjihrig) [inline lib/setup.js:4]: That's just the hapi style ;-)

**Pull request closed without merging**